src/Sarif/UriHelper.cs

```diff
@@ -69,6 +71,16 @@ public static string MakeValidUri(string path)
                     !path.StartsWith(@"\", StringComparison.Ordinal))
                 {
                     validUri = validUri.Substring(1);
+
+                    // When the UriBuilder constructs an absolute URI, it strips any
+                    // leading "." and ".." segments ("dot-segments", as RFC 3986 calls
+                    // them). Glue them back on so we don't lose the relative path
+                    // information.
+                    string leadingDotSegments = GetLeadingDotSegments(path);
+                    if (!string.IsNullOrEmpty(leadingDotSegments))
+                    {
+                        validUri = leadingDotSegments + validUri;
+                    }
```

Comment:
Well, I'm probably wrong about that second issue. I just know I reworked the logic a bit in a way I fondly believed made it a bit simpler :-)